### ISSUE TITLE
Transfer items will transfer items in non-interactable containers.

### ISSUE THREAD

**Issue by esirprus** [Regalis11/Barotrauma#9381]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

![image](https://user-images.githubusercontent.com/44423036/174216482-f251f1c4-78fd-4b14-b0b9-097d166b68b7.png)
![image](https://user-images.githubusercontent.com/44423036/174216492-f869e3c9-e143-4ca9-a46f-27db219733b7.png)
barsuk(just edit an ammunition shelf to non-interactable) to humpback

### Reproduction steps

1. edit the ammunition shelf to non interactable, and fill with pulse tri-laser fuel box(Convenient to observe)
2. start a campaign and choose your edited sub
3. buy and switch to a new sub, choose transfer items
4. see pulse tri-laser fuel box is on your new sub

### Bug prevalence

Happens every time I play

### Version

0.18.11.0

### Which operating system did you encounter this bug on?

Windows

**Comment by esirprus** [comment]
if you use weapon holder and an item to decorate your sub (this is very common) and you forget to set the item non-interactable, this will happen when you change a new sub

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/343af06b385c24bd167f9f466b9c16a528285171

**Comment by esirprus** [comment]
![unknown](https://user-images.githubusercontent.com/44423036/176582458-d4fe1248-d52b-45a3-b19d-76528218af17.png)
This problem was encountered by a player on discord after changing to a new submarine and I wanted to confirm if this problem was related to my issue. There are also claims that transferring items when replacing the submarine will transfer the non-interactable ammo boxes from the drones.

**Comment by NilanthAnimosus** [comment]
Tested against bugfixes commit https://github.com/Regalis11/Barotrauma-development/commit/6efb667a91d39ee7715a094a89a32ed20b73d6ec, needs further looking into.

Items no longer transfer out of or to non-interactable containers. I also tried having one full and one empty set of ammunition racks on a shuttle in my test to see if those would transfer (Both not interactable) and I couldn't get anything to be added or removed from said inventories...and nothing was placed into a deconstructor or something (Though it is all tags based so a custom submarine could make that happen)?

I have noticed one important detail however - If you set something non-interactable yet then proceed to forget to remove the autofill, it will get filled with new items on a campaign start but I would call this intentional behavior personally as a sub creator may want random items placed into a none-interactable (Like different kinds of diving suit or ammo boxes or something visually).

As for the guardian steam cannon - it appears spawning this in starts it off as "Locked" when spawned which denies any interaction of whatever inventory its in, even if spawned straight to the players own inventory. I doubt anyone could place it into the deconstructor as a result (you cant spawnitem it into there in my experience) so I believe it did likely transfer a (Locked) state item, which should probably be prevented from being allowed.
